### PR TITLE
Adding support for email

### PR DESCRIPTION
**Pull request by wataylor88**
- Added code to email when daisy troop reservation is made
- Added code to email on booth deletion
- Refactoring of users/models.py to fix circular dependency

Testing:
Manual testing shows emails working ok, existing ticket max functionality working. Unit tests passing.

### CODE REVIEW

File: cookie_booths/models.py [hunk 1]
```diff
@@ -507,5 +512,30 @@ def generate_hours_if_needed(sender, instance, created, **kwargs):
...
+    if not reserved_troop:
+        return
+
+    # Then the owner from that
+    block_owner = User.objects.get(username=reserved_troop.troop_cookie_coordinator)
+    if not block_owner:
+        return
+
+    title = "Your Reserved Booth Block Has Been Removed"
+    message = "Hello " + block_owner.first_name + ",\n" + \
```
Comment (wataylor88) [inline cookie_booths/models.py:530]: We can tweak this to whatever but I thought this was a good first crack at it.

File: cookie_booths/views.py [hunk 2]
```diff
@@ -324,14 +326,50 @@ def is_block_reserved_by_user(request, block_id):
...
 
 
+def get_week_start_end_from_date(date):
+    start_date = date - timedelta(days=date.weekday())
+    end_date = start_date + timedelta(days=6)
+
+    return start_date, end_date
+
+
+def get_num_tickets_remaining(troop, date):
```
Comment (wataylor88) [inline cookie_booths/views.py:336]: Basically copied from users/models.py - moved here to remove circular dependency so cookie_booths/models.py has access to Troop/User for email purposes.

File: cookie_website/settings.py [hunk 3]
```diff
@@ -139,3 +139,11 @@
 
 # Heroku settings
 django_heroku.settings(locals())
+
+# Email settings
```
Comment (wataylor88) [inline cookie_website/settings.py:143]: The website I had recommended adding the HOST/USER/PASSWORD to an environment file to obscure it, but I know you also need to do the secret key as well, so I just left these in here for now.

**Review by wataylor88 — COMMENTED** [review]
Some screenshots of the emails sent from the site email, to my personal email, sending and receiving just fine.

blocktimechange
daisyconfirmation

File: cookie_booths/views.py (hunk 2 above)
Comment (mshappel) [reply cookie_booths/views.py:336]: We may want to consider creating a module for these items that don't really have a good home. Nothing that needs to change now.

**Review by mshappel — APPROVED** [review]

**Pull request merged**